docs/adrs/0008-appliance-calculator-location.md

```diff
@@ -0,0 +1,83 @@
+# 8. Appliance Calculator Code Location
+
+Date: 2024-10-15
+
+## Status
+
+Draft, Proposed
+
+## Context
+
+The energy team's appliance calculator is currently served be EpiServer at https://www.citizensadvice.org.uk/consumer/energy/energy-supply/save-energy-at-home/check-how-much-your-electrical-appliances-cost-to-use/.  
+
+The Digital Energy Team is rebuilding this app as a progressively enhanced Rails app and we need to decide where the code for that app should live.  
+
+The energy team currently have three code repos for their apps in production:
+- [citizensadvice/energy-prototypes](https://www.github.com/citizensadvice/energy-prototypes) (current appliance calculator, will be archived on completion of this work)
+- [citizensadvice/smart-meter-tool](https://github.com/citizensadvice/smart-meter-tool) (has specific contractual deployment and hosting isolation requirements)
+- [citizensadvice/energy-comparison-table](https://github.com/citizensadvice/energy-comparison-table) (single rails app handling requests for the customer service rating comparison table)
+
+The appliance calculator itself will be rendered into an iframe on an advice collection page by the public website, which involves creating a [new content type](https://app.contentful.com/spaces/vms0u05139aw/environments/qa/content_types/interactiveTool/fields) to represent the calculator, making content-api understand that new content type and creating a new renderer in the public website codebase.  Rough implementation spikes have proved that this will work:
+- https://github.com/citizensadvice/content-api/tree/exp/interactive-tool-spike
+- https://github.com/citizensadvice/public-website/tree/exp/interactive-tool-spike
+
+## Decision
+
+1. Generalise from `energy-comparison-table` to `energy-apps`, by:
+   1. recreating the current app infrastructure with the more generic `energy-apps` name
+   2. switching to the the new sub-domain in production in the CloudFront config
+   3. renaming the `energy-comparison-table` app and namespacing the comparison table app appropriately
+2. Build the appliance calculator in the newly renamed `energy-apps` rails app
+
+## Consequences
+
+### Positive
+
+- all energy apps (excluding the smart meter tool) are contained in the same place
+- we will only need to solve certain problems once for all energy apps (eg monitoring, authentication, header and footer, country route handling)
+- little to no additional maintenance burden
+- can leverage the existing build and deployment workflows (which are robust and fast)
+- increased exposure to CDK and infrastructure configration for the energy team developer (the current infrastructure code was developed by the SRE team)
+
+
+### Negative
+
```

Comment:
Is there a possibility that we might want to create other interactive tools in the future? Would these necessarily be related to energy?

Reply:
There is!  And if we did, we could use the same content type and keep the app code that serves the iframe where ever made the most sense